app/controllers/ems_common.rb

```diff
@@ -427,27 +427,66 @@ def update_button_reset
 
   def update_button_validate
     verify_ems = find_by_id_filtered(model, params[:id])
-    validate_credentials verify_ems
+    queue_credential_validation_task verify_ems
   end
   private :update_button_validate
 
-  def validate_credentials(verify_ems)
+  def queue_credential_validation_task(verify_ems)
     set_record_vars(verify_ems, :validate)
+    creds = set_record_creds(verify_ems)
+
     @in_a_form = true
     @changed = session[:changed]
 
     # validate button should say "revalidate" if the form is unchanged
     revalidating = !edit_changed?
-    result, details = verify_ems.authentication_check(params[:type], :save => revalidating)
-    if result
-      add_flash(_("Credential validation was successful"))
+
+    args = {
+      :ems        => verify_ems,
+      :creds      => creds,
```

Comment:
Yeah, it feels like we're jumping through lots of hoops just because we don't want to save a record in the database for the UI.  Note, this isn't the first time this in memory new Ems that can't be saved was a problem.  I just don't understand what we gain from it in the UI to warrant the hacks we have to do in the backend to not preserve the object.